File: 06-special-class-members/code_from_slides/triangles/RightTriangle.java
public class RightTriangle {
    private Double a;
    private Double b;
    public RightTriangle(Double a, Double b) {
        this.a = a;
        this.b = b;
    }
    public Double hypotenuse() {
        Double c = Math.sqrt(a*a + b*b);
        return c;
    }
    public static void main(String[] args) {
        Double a = 3.0;
        Double b = 4.0;
        RightTriangle rt = new RightTriangle(a, b);
        Double c = rt.hypotenuse();
        System.out.printf("The hypotenuse is: %f\n", c);
    }
}

File: 06-special-class-members/code_from_slides/triangles/right_triangle.c
// build using gcc right_triangle.c -lm
// run using ./a.out

#include <stdio.h>
#include <stdlib.h>
#include <math.h>

double* hypotenuse(double *a, double *b) {
    double *c = malloc(sizeof(double));
    *c = sqrt((*a)*(*a) + (*b)*(*b));
    return c;
}
  
void main() {
    double a = 3;
    double b = 4;
    double *c = hypotenuse(&a, &b);
    printf("The hypotenuse is: %f\n", *c);
    free(c);
}
